include/upipe-dvb/upipe_dvbsrc.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2017 Open Broadcast Systems Ltd
+ *
+ * Author: Rafaël Carré
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston MA 02110-1301, USA.
+ */
+
+/** @file
+ * @short Upipe source module for dvb receivers
+ */
+
+#ifndef _UPIPE_MODULES_UPIPE_DVB_SOURCE_H_
+/** @hidden */
+#define _UPIPE_MODULES_UPIPE_DVB_SOURCE_H_
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <upipe/upipe.h>
+
+#define UPIPE_DVBSRC_SIGNATURE UBASE_FOURCC('d','v','b',' ')
+
+enum upipe_dvbsrc_command {
+    UPIPE_DVBSRC_SENTINAL = UPIPE_CONTROL_LOCAL,
+
+    UPIPE_DVBSRC_GET_FRONTEND_STATS, /* int sig, unsigned int *, float *, unsigned int *, float * */
+};
+
+static inline int upipe_dvbsrc_get_frontend_stats(struct upipe *upipe,
+        unsigned int *status, float *signal, unsigned int *signal_db, float *snr)
+{
```

Comment:
You should consider that this list may evolve in the future. I'd like to avoid upipe_dvbsrc_get_frontend_stats2 and stats3 when you add eg. uncorrected blocks, ber, etc. Use a struct ? A variadic function terminated by NULL ?